src/storage/v2/durability/snapshot.cpp

```diff
@@ -2966,6 +3467,46 @@ RecoveredSnapshot LoadSnapshot(const std::filesystem::path &path, utils::SkipLis
       spdlog::info("Metadata of point indices are recovered.");
     }
 
+    // Recover vector indices.
+    {
+      auto size = snapshot.ReadUint();
+      if (!size) throw RecoveryFailure("Couldn't recover the number of vector indices!");
+      spdlog::info("Recovering metadata of {} vector indices.", *size);
+      for (uint64_t i = 0; i < *size; ++i) {
+        auto index_name = snapshot.ReadString();
+        if (!index_name.has_value()) throw RecoveryFailure("Couldn't read vector index name!");
+        auto label = snapshot.ReadUint();
+        if (!label) throw RecoveryFailure("Couldn't read vector index label!");
+        auto property = snapshot.ReadUint();
+        if (!property) throw RecoveryFailure("Couldn't read vector index property!");
+        auto metric = snapshot.ReadString();
+        if (!metric) throw RecoveryFailure("Couldn't read vector index metric!");
+        auto usearch_metric = unum::usearch::metric_from_name(metric->data(), metric->size());
+        if (usearch_metric.error) throw RecoveryFailure("Invalid vector index metric recovered!");
+        auto dimension = snapshot.ReadUint();
+        if (!dimension) throw RecoveryFailure("Couldn't read vector index dimension!");
+        auto capacity = snapshot.ReadUint();
+        if (!capacity) throw RecoveryFailure("Couldn't read vector index capacity!");
+        auto resize_coefficient = snapshot.ReadUint();
+        if (!resize_coefficient) throw RecoveryFailure("Couldn't read vector index resize coefficient!");
+        auto spec = std::make_shared<VectorIndexSpec>(
+            std::move(index_name.value()), get_label_from_id(*label), get_property_from_id(*property), usearch_metric,
+            static_cast<std::uint16_t>(*dimension), *capacity, static_cast<std::uint16_t>(*resize_coefficient));
+
+        if (std::ranges::any_of(indices_constraints.indices.vector_indices, [&spec](const auto &vector_index) {
```

Comment:
move to before building a spec, this way we can directly validate against `index_name` before moving into the spec